features/landscape.feature

```diff
@@ -0,0 +1,285 @@
+        Updating package lists
+        Installing landscape-client
+        Executing `landscape-config --computer-title $behave_var{machine-name system-under-test} --account-name pro-client-qa --silent`
+        Created symlink /etc/systemd/system/multi-user.target.wants/landscape-client.service → /lib/systemd/system/landscape-client.service.
+        Invalid account name or registration key.
+        Could not enable Landscape.
+        """
+        When I run `pro status` with sudo
+        Then stdout matches regexp:
+        """
+        landscape +yes +warning
+        """
+        Then stdout contains substring:
+        """
+        Landscape is installed and configured but not registered.
+        Run `sudo landscape-config` to register, or run `sudo pro disable landscape`
+        """
+        When I run `sudo pro disable landscape` with sudo
+
+        # Enable with assume-yes and format json
+        When I run `pro enable landscape --assume-yes --format=json -- --computer-title $behave_var{machine-name system-under-test} --account-name pro-client-qa --registration-key $behave_var{config landscape_registration_key}` with sudo
+        Then I will see the following on stdout:
+        """
+        {"_schema_version": "0.1", "errors": [], "failed_services": [], "needs_reboot": false, "processed_services": ["landscape"], "result": "success", "warnings": []}
+        """
+        When I run `pro status` with sudo
+        Then stdout matches regexp:
+        """
+        landscape +yes +enabled
+        """
+        When I run `sudo pro disable landscape` with sudo
+
+        # Fail to enable with assume-yes and format json
+        When I verify that running `pro enable landscape --assume-yes --format=json -- --computer-title $behave_var{machine-name system-under-test} --account-name pro-client-qa` `with sudo` exits `1`
+        Then I will see the following on stdout:
+        """
+        {"_schema_version": "0.1", "errors": [{"additional_info": {"stderr": "Created symlink /etc/systemd/system/multi-user.target.wants/landscape-client.service \u2192 /lib/systemd/system/landscape-client.service.\nInvalid account name or registration key.", "stdout": "Please wait..."}, "message": "landscape-config command failed", "message_code": "landscape-config-failed", "service": "landscape", "type": "service"}], "failed_services": ["landscape"], "needs_reboot": false, "processed_services": [], "result": "failure", "warnings": []}
+        """
+        When I run `pro status` with sudo
+        Then stdout matches regexp:
+        """
+        landscape +yes +warning
+        """
+        Then stdout contains substring:
+        """
+        Landscape is installed and configured but not registered.
+        Run `sudo landscape-config` to register, or run `sudo pro disable landscape`
+        """
+        When I run `sudo pro disable landscape` with sudo
+
+        # cleanup
+        Then I reject all pending computers on Landscape
```

Comment:
I'd specify `on the Landscape server` here for explicity

Reply:
I think I'm going to move this to a "after_all" hook or something so that it cleans up even if the test fails

Reply:
will do this in a followup: #2676